artiq/coredevice/mirny.py

```diff
@@ -140,3 +151,232 @@ def write_ext(self, addr, length, data):
         if length < 32:
             data <<= 32 - length
         self.bus.write(data)
+
+
+class Almazny:
+    """
+    Almazny (High frequency mezzanine board for Mirny)
+
+    :param host_mirny - Mirny device Almazny is connected to
+    """
+
+    def __init__(self, dmgr, host_mirny):
+        self.mirny = dmgr.get(host_mirny)
+        self.att_mu = [0x3f] * 4
+        self.rf_switches = [0] * 4
+
+    @kernel
+    def init(self):
+        self._send_mezz_data([
+            (ALMAZNY_REG_CLEAR, 1),
+            # rest is 0
+            ])
+
+    @kernel
+    def _reverse_att_mu(self, att_mu):
+        res = 0
+        for i in range(6):
+            res |= (att_mu >> i & 0x01) << (5-i)
+        return res
+
+    @kernel
+    def att_to_mu(self, att):
+        """
+        Convert an attenuator setting in dB to machine units.
+
+        :param att: attenuator setting in dB [0-31.5] in 0.5 steps
+        :return: attenuator setting in machine units
+        """
+        if att > 31.5 or att < 0 or (att-int(att) != 0.0 and att-int(att) != 0.5):
```

Comment:
Remove the ``.5`` check, rounding is always present one way or another and generally acceptable.